ammico/display.py

```diff
@@ -157,30 +168,68 @@ def _right_output_json(self):
         return right_layout
 
     def run_server(self, port=8050):
+        """
+        Runs the Dash server to start the analysis explorer.
+
+        Parameters:
+        - port: The port number to run the server on (default: 8050).
+
+        Note:
+        - This method should only be called in an interactive environment like Jupyter notebooks.
+        - Raises an EnvironmentError if not called in an interactive environment.
+        """
         if not is_interactive():
             raise EnvironmentError(
-                "Dash server should only be called in interactive an interactive environment like jupyter notebooks."
+                "Dash server should only be called in an interactive environment like Jupyter notebooks."
             )
 
         self.app.run_server(debug=True, mode="inline", port=port)
 
     # Dash callbacks
     def update_picture(self, img_path):
+        """
+        Callback function to update the displayed image.
+
+        Parameters:
+        - img_path: The path of the selected image.
+
+        Returns:
+        - The image object to be displayed.
+
+        Note:
+        - This function is called when the value of the file explorer dropdown changes.
+        - Reads the image file and returns the image object.
+        """
         if img_path is not None:
```

Comment:
Why should it be None? It has to be provided for this method. Better would be to check if an image is found at that path and return  a sensible  error message if not.

Reply:
because the function is used as a Dash callback and these can sometimes be triggered without an argument. 
Technically the `prevent_initial_call=True` argument of the callback declaration should prevent this, but I'm never 100% sure dash doesn't do some shenanigans in the background.